### PR TITLE
Release 2.2.0

### PR DESCRIPTION
**Pull request by George3d6**

### CODE REVIEW

File: mindsdb_native/libs/phases/type_deductor/type_deductor.py [hunk 1]
```diff
@@ -198,8 +202,27 @@ def get_column_data_type(self, data, full_data, col_name):
...
+                try:
+                    item_tags = [t.strip() for t in item.split(',')]
+                    lengths.append(len(item_tags))
+                    unique_tokens = unique_tokens.union(set(item_tags))
+                except AttributeError:
+                    can_be_tags = False
+                    break
+
+            # If more than 30% of the samples contain more than 1 category and there's more than 10 of them and they are shared between the various cells
+            if can_be_tags and np.mean(lengths) > 1.3 and len(unique_tokens) >= 6 and len(unique_tokens)/np.mean(lengths) < (len(data)/4):
```
Comment (paxcema) [inline mindsdb_native/libs/phases/type_deductor/type_deductor.py:219]: Is the final criterion 6 or 10 unique categories?

File: mindsdb_native/libs/phases/type_deductor/type_deductor.py [hunk 2]
```diff
@@ -198,8 +202,27 @@ def get_column_data_type(self, data, full_data, col_name):
...
             curr_data_type, curr_data_subtype = None, None
 
+        # Check for Tags subtype
+        if curr_data_subtype != DATA_SUBTYPES.ARRAY:
+            lengths = []
+            unique_tokens = set()
+            can_be_tags = True
+            for item in data:
+                try:
+                    item_tags = [t.strip() for t in item.split(',')]
```
Comment (maximlopin) [inline mindsdb_native/libs/phases/type_deductor/type_deductor.py:214]: the `try except` can be replaced with a simple `if` statement:
```
if isinstance(item, str):
    item_tags = [t.strip() for t in item.split(',')]
    lengths.append(len(item_tags))
    unique_tokens = unique_tokens.union(set(item_tags))
else:
    can_be_tags = False
    break
```

File: tests/unit_tests/libs/phases/type_deductor/test_type_deductor.py [hunk 3]
```diff
@@ -52,6 +54,8 @@ def test_type_deduction(self, transaction, lmd):
         # Apparently for n_category_values = 10 it doesnt work
         n_category_values = 4
         categories_cycle = cycle(range(n_category_values))
+        n_multilabel_category_values = 25
+        multiple_categories_str_cycle = cycle([random.choice(VOCAB[0:20]) for i in range(n_multilabel_category_values)])
```
Comment (maximlopin) [inline tests/unit_tests/libs/phases/type_deductor/test_type_deductor.py]: Can be simplified to:
```
cycle(
    random.choices(VOCAB[0:20], k=n_multilabel_category_values)
)
```

File: tests/unit_tests/libs/controllers/test_predictor.py [hunk 4]
```diff
@@ -22,7 +23,8 @@
                                     generate_value_cols,
                                     generate_timeseries_labels,
                                     generate_log_labels,
-                                    columns_to_file, PickableMock)
```
Comment (maximlopin) [inline tests/unit_tests/libs/controllers/test_predictor.py:25]: ugly indentation

File: mindsdb_native/libs/phases/type_deductor/type_deductor.py (hunk 1 above)
Comment (George3d6) [reply mindsdb_native/libs/phases/type_deductor/type_deductor.py:219]: 6

File: mindsdb_native/libs/phases/type_deductor/type_deductor.py (hunk 2 above)
Comment (George3d6) [reply mindsdb_native/libs/phases/type_deductor/type_deductor.py:214]: Noted, simplified this even further.

File: tests/unit_tests/libs/phases/type_deductor/test_type_deductor.py (hunk 3 above)
Comment (George3d6) [reply tests/unit_tests/libs/phases/type_deductor/test_type_deductor.py]: done

**Pull request merged**